R/Rules-methods.R

```diff
@@ -4,127 +4,133 @@
-##' on the grid defined in \code{data}, and a plot depicting this recommendation
-##' (element \code{plot}). In case of multiple plots also an element \code{singlePlots}
-##' is included which returns the list of single plots, which allows for further
-##' customization of these. Also additional list elements describing the outcome
-##' of the rule can be contained.
-##'
-##' @export
-##' @keywords methods
-setGeneric("nextBest",
-           def=
-               function(nextBest, doselimit, samples, model, data, ...){
-
-                   ## there should be no default method,
-                   ## therefore just forward to next method!
-                   standardGeneric("nextBest")
-               },
-           valueClass="list")
-
-## --------------------------------------------------
-## The MTD method
-## --------------------------------------------------
-
-##' @describeIn nextBest Find the next best dose based on the MTD rule
-##'
-##' @example examples/Rules-method-NextBestMTD.R
-##' @importFrom ggplot2 ggplot geom_density xlab ylab xlim aes geom_vline
-##' geom_text
-setMethod("nextBest",
-          signature=
-          signature(nextBest="NextBestMTD",
-                    doselimit="numeric",
-                    samples="Samples",
-                    model="Model",
-                    data="Data"),
-          def=
-              function(nextBest, doselimit, samples, model, data, ...){
-
-                  if(identical(length(doselimit), 0L))
-                  {
-                      warning("doselimit is empty, therefore no dose limit will be applied")
-                  }
-
-              ## First, generate the MTD samples.
-              mtd_samples <- dose(x=nextBest@target,
-                                 model,
-                                 samples)
+# nextBest ----
 
-              ## then derive the next best dose
-              mtdEstimate <- nextBest@derive(mtd_samples=mtd_samples)
+## generic ----
 
-              ## be sure which doses are ok with respect to maximum
-              ## possible dose - if one was specified
-              dosesOK <-
-                  if(length(doselimit))
-                      which(data@doseGrid <= doselimit)
-                  else
-                      seq_along(data@doseGrid)
+#' Finding the Next Best Dose
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A function that computes the recommended next best dose based on the
+#' corresponding rule `nextBest`, the posterior `samples` from the `model` and
+#' the underlying `data`.
+#'
+#' @param nextBest (`NextBest`)\cr the rule for the next best dose.
+#' @param doselimit (`number`)\cr the maximum allowed next dose. If it an empty
+#'   vector, then no dose limit will be applied in the course of dose
+#'   recommendation calculation, and a corresponding warning is given.
+#' @param samples (`Samples`)\cr a samples.
+#' @param model (`Model`)\cr model input.
+#' @param data (`Data`)\cr data input.
```

Comment:
```suggestion
#' @param model (`Model`)\cr used model.
#' @param data (`Data`)\cr used data.
```
(just to allude to that this was used to generate the samples)